eclipse-theia/theia
# Hide the sash when the left/right side panel is collapsed and when the horizontal container is collapsed

#### What it does

- Hide the sash when the left/right side panel is collapsed (see the two videos for comparison)


https://user-images.githubusercontent.com/23107734/146595893-a428f02c-6b55-49bf-b88c-ca63b186d0ae.mov


https://user-images.githubusercontent.com/23107734/146595902-f8cbfb98-b8a3-4410-9321-75b3d604e148.mov


- Fix bug for the scenario in the left panel where the sash is visible when all horizontal containers are collapsed (please refer to the screenshot below for more details)

| Before  | After | 
| ------------- | ------------- | 
| ![image](https://user-images.githubusercontent.com/23107734/147305518-0e1f6e71-b1db-4fb0-b5e3-ce9094bcd53e.png)  | ![image](https://user-images.githubusercontent.com/23107734/147306300-ba58b3b9-2270-471a-96f1-d776047a992a.png)  |



#### How to test
1. Verify that the sash is hidden when the left/right side panel is collapsed
2. Verify that the sash for the left/right side panel functions properly (try updating its size and hoverDelay)
3. Install `GitLens` or `Maven for Java` extension from the VSX Registry (GitLens is used in the screenshot above)
4. Locate a horizontal container for the extension from the left panel. Without clicking on the horizontal container, verify that the sash is hidden when the container is collapsed

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
